src/compiler/binder.ts

```diff
@@ -2558,11 +2571,18 @@ namespace ts {
             node.left.parent = node;
             node.right.parent = node;
             const lhs = node.left as PropertyAccessEntityNameExpression;
-            bindPropertyAssignment(lhs.expression, lhs, /*isPrototypeProperty*/ false);
+            const constructorSymbol = lookupSymbolForPropertyAccess(lhs.expression);
+            if (constructorSymbol) {

```

Comment:
I think this binding should be unconditional, so a merge cross-file can occur. Eg,
```js
// file1.js
function C() {
  this.a = 2;
}
// file2.js
C.prototype.foo = function() { return this.a; };
```

Reply:
This is pretty complicated because of things like

```js
Outer.Inner = function() { this.a = 1 }
Outer.Inner.prototype = { ... }
```

Probably needs an iterated declareSymbol just like declarePossiblyMissingNamespace.

Reply:
Yep, that sounds about right.